drivers/sdhc/rcar_mmc.c

```diff
@@ -259,7 +258,7 @@ static int rcar_mmc_poll_reg_flags_check_err(const struct device *dev,
 		}
 
 		k_usleep(MMC_POLL_FLAGS_ONE_CYCLE_TIMEOUT_US);
-		remain_us -= MMC_POLL_FLAGS_ONE_CYCLE_TIMEOUT_US;
+		timeout_us -= MMC_POLL_FLAGS_ONE_CYCLE_TIMEOUT_US;
```

Comment:
MISRA C frowns upon changing values of function parameters (rule 17.8). I do understand that this rule is advisory and that there are plenty of places where this rule is violated. But in your case it was easier to follow it,  because resulting diff would be smaller. So, you if want, you can return back `remain_us`, just initialize it with `timeout_us`.

Reply:
some of calls use int32 some of calls use int literal constant
even if constant will be less then int32 (which is impossible according this comment https://github.com/xen-troops/zephyr/pull/47#discussion_r1210014393), literal constant will be promoted to higher int32 value with sign propagation

so if you don't have a crucial reason to change this code, I leave it change as is

Reply:
Well, strictly speaking, this is not a variable, but I got your point.